src/azure-cli/azure/cli/command_modules/appconfig/custom.py

```diff
@@ -49,8 +50,27 @@ def create_configstore(client,
                                             sku=Sku(name=sku),
                                             tags=tags,
                                             public_network_access=public_network_access,
-                                            disable_local_auth=disable_local_auth)
+                                            disable_local_auth=disable_local_auth,
+                                            soft_delete_retention_in_days=retention_days,
+                                            enable_purge_protection=enable_purge_protection,
+                                            create_mode=CreateMode.DEFAULT)
+
+    return client.begin_create(resource_group_name, name, configstore_params)
+
 
+def recover_deleted_configstore(cmd, client, name, resource_group_name=None, location=None, yes=False):
+    if resource_group_name is None or location is None:
+        metadata_resource_group, metadata_location = resolve_deleted_store_metadata(cmd, name, resource_group_name, location)
+
+        if resource_group_name is None:
+            resource_group_name = metadata_resource_group
+        if location is None:
+            location = metadata_location
+
+    configstore_params = ConfigurationStore(location=location.lower(),
+                                            sku=Sku(name="Standard"),  # Only Standard SKU stores can be recovered!
```

Comment:
Saw a similar comment by Shuai but can't find it now. What happens if someone is trying to run this command on a free store and we hard code the sku to standard? Will the server throw an error because the hardcoded sku does not match the real sku of this store?

Reply:
Any store can only be recovered if it is within the retention period. The retention period for a free store is 0 days. Since free store won't be retained, executing this command on a free store would fail anyways because you can't recover permanently deleted store.

Reply:
Enabling purge protection on a free store is not failing. The command succeeds as though purge protection was successfully enabled. Even though the retention period is 0 days, it gives the illusion that purge protection has been enabled on a free store.

Reply:
I think that is something the server should validate.

Reply:
I'm a bit hesitant to add redundant validations (adding same validations as the server) in CLI because it tightens the coupling.
If something changes on the server, we land in an inconsistent state and we need to update the CLI client again.

Reply:
> What I meant was that RP should do this validation so that CLI doesn't have to. But in case RP cannot validate it (maybe some implementation detail on RP side), this error cannot be displayed as it is to users. It doesn't make any sense to the user. In that case, we need to handle the error in CLI as a last resort.

That totally makes sense.

> Server changes will not be reflected to CLI users until we decide to upgrade the CLI to use a newer SDK.
Ah, yes. Thank you for reminding me. I totally forgot that the changes won't be reflected unless we upgrade the SDK.